rdkit/Chem/AtomPairs/Torsions.py

```diff
@@ -54,33 +54,27 @@ def pyScorePath(mol, path, size, atomCodes=None):
   >>> c3 = Utils.GetAtomCode(m.GetAtomWithIdx(2),2)
   >>> c4 = Utils.GetAtomCode(m.GetAtomWithIdx(4),1)
   >>> t = c1 | (c2 << rdMolDescriptors.AtomPairsParameters.codeSize) | (c3 << (rdMolDescriptors.AtomPairsParameters.codeSize*2)) | (c4 << (rdMolDescriptors.AtomPairsParameters.codeSize*3))
-  >>> pyScorePath(m,(0,1,2,4),4)==t
+  >>> pyScorePath(m, (0, 1, 2, 4), 4) == t
   1
 
   """
   codes = [None] * size
   for i in range(size):
-    if i == 0 or i == (size - 1):
-      sub = 1
-    else:
-      sub = 2
+    sub = 1 if i in (0, size - 1) else 2
     if not atomCodes:
       codes[i] = Utils.GetAtomCode(mol.GetAtomWithIdx(path[i]), sub)
     else:
-      base = atomCodes[path[i]]
-      codes[i] = base - sub
+      codes[i] = atomCodes[path[i]] - sub
 
   # "canonicalize" the code vector:
-  beg = 0
-  end = len(codes) - 1
-  while (beg < end):
-    if codes[beg] > codes[end]:
-      codes.reverse()
-      break
-    elif codes[beg] == codes[end]:
+  beg, end = 0, len(codes) - 1
+  while beg < end:
+    if codes[beg] == codes[end]:
       beg += 1
       end -= 1
     else:
+      if codes[beg] > codes[end]:
+        codes.reverse()
       break
   accum = 0
   for i in range(size):
```

Comment:
In the following loop
```
  accum = 0
  for i in range(size):
    accum |= codes[i] << (rdMolDescriptors.AtomPairsParameters.codeSize * i)
```
could use `enumerate` here
```
  accum = 0
  for i, code in enumerate(codes):
    accum |= code << (rdMolDescriptors.AtomPairsParameters.codeSize * i)
```
Also looks like `rdMolDescriptors.AtomPairsParameters.codeSize` is a constant, so could be moved out of the loop. This might be a micro-optimisation that isn't worth doing though, but would make it clearer what is being done here.
```
  accum = 0
  codeSize = rdMolDescriptors.AtomPairsParameters.codeSize
  for i, code in enumerate(codes):
    accum |= code << (codeSize * i)
```